src/main/flight/autopilot.c

```diff
@@ -91,17 +187,183 @@ void altitudeControl(float targetAltitudeCm, float taskIntervalS, float vertical
     DEBUG_SET(DEBUG_AUTOPILOT_ALTITUDE, 7, lrintf(altitudeF));
 }
 
-bool isBelowLandingAltitude(void)
+void setSticksActiveStatus(bool areSticksActive)
 {
-    return getAltitudeCm() < 100.0f * autopilotConfig()->landing_altitude_m;
+    posHold.sticksActive = areSticksActive;
 }
 
-const pidCoefficient_t *getAltitudePidCoeffs(void)
+void setTargetLocation(gpsLocation_t newTargetLocation)
 {
-    return &altitudePidCoeffs;
+    currentTargetLocation = newTargetLocation;
+    posHold.efAxis[EW].previousDistance = 0.0f; // reset to avoid D and A spikess
+    posHold.efAxis[NS].previousDistance = 0.0f;
+    // function is intended for only small changes in position
+    // for example, where the step distance change reflects an intended velocity, determined by a client function
+    // if we had a 'target_ground_speed' value, like in gps_rescue, we can make a function that starts and stops smoothly and targets that velocity
+}
+
+void resetLocation(earthFrame_t *efAxis, axisEF_t loopAxis)
+{
+    if (loopAxis == EW) {
+        currentTargetLocation.lon = gpsSol.llh.lon; // update East-West / / longitude position
+    } else {
+        currentTargetLocation.lat = gpsSol.llh.lat; // update North-South / latitude position
+    }
+    efAxis->previousDistance = 0.0f; // and reset the previous distance
+}
+
+bool positionControl(void) 
+{
+    static uint16_t previousGpsStamp = ~0;
+    if (currentGpsStamp() != previousGpsStamp) {
+        previousGpsStamp = currentGpsStamp();
```

Comment:
IMO implementing it this way is cleaner. gpsStamp is opaque from outside, it it only used inside GPS code to decide.
I'll create PR with implementation
```suggestion
    static uint16_t gpsStamp = ~0;
    if (gpsNewData(&gpsStamp) {
```

Reply:
That's fantastic, thank you.  Let me check it and I'll add you suggested detailed method some time in the next 48 hours.

Reply:
can I add this into the upcoming PR that includes GPS rescue, where I have tested it already?